test/parallel/test-repl-history-perm.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+// Flags: --expose_internals
+
+const common = require('../common');
+
+if (common.isWindows) {
+  console.log('1..0 # Skipped: Win32 uses ACLs for file permissions, ' +
+              'modes are always 0666 and says nothing about group/other ' +
+              'read access.');
+  return;
+}
+
+const assert = require('assert');
+const path = require('path');
+const fs = require('fs');
+const repl = require('internal/repl');
+const Duplex = require('stream').Duplex;
+// Invoking the REPL should create a repl history file at the specified path
+// and mode 600.
+
+var stream = new Duplex();
+stream.pause = stream.resume = function() {};
+// ends immediately
+stream._read = function() {
+  this.push(null);
+};
+stream._write = function(c, e, cb) {
+  cb();
+};
+stream.readable = stream.writable = true;
+
+common.refreshTmpDir();
+const replHistoryPath = path.join(common.tmpDir, 'repl_history');
+
+const checkResults = common.mustCall(function(err, r) {
+  if (err)
+    throw err;
+  r.input.end();
+  const stat = fs.statSync(replHistoryPath);
+  const mode = '0' + (stat.mode & parseInt('777', 8)).toString(8);
+  assert.strictEqual(mode, '0600', 'REPL history file should be mode 0600');
```

Comment:
Since `stat.mode` comes back as something like  `0o100600` you'd still need to do masking:

    assert.strictEqual(stat.mode & 0o777, 0o600);

If you're goal is legibility, you may be best off converting to a string:

    const mode = stat.mode.toString(8);
    assert(mode.endsWith('600'));